### PR TITLE
[Specification 2.0]: Add entry about headings

### PR DESCRIPTION
**Pull request by vhyrro**
This PR adds a headings section and attempts to describe them in the most succinct way possible.

### CODE REVIEW

**Comment by boltlessengineer** [comment]
Can we rename _heading_ to _section_? I think it makes more sense.

heading -> section
heading title -> heading

So when we say "content of heading" we explicitly say for the text coming after the asterisks.

This will also help users from markdown-like syntax to understand how heading in Norg work differently. (heading _indents_ all the content below)

**Comment by vhyrro** [comment]
I don't see any added value of changing the definitions here honestly. Since norg is more org-derived than markdown-derived I'd keep the original term and not break the existing established wording in the v1 spec. "Title" is plenty explanatory :)

**Comment by boltlessengineer** [comment]
I'm saying with the context of #43.
If we want to put heading everywhere and define a "block" node, heading itself should not do anything structural.

```
- * heading
** subheading
- second list item??
```
This will be hard to parse, way more than before.

If heading can be anywhere as a block **without** structuring the document, we are actually allowing _title_, not _heading_.

Then the wording is bit confusing because now we have to say "title can be placed anywhere that block item can exist".
The word "title" doesn't seem to fit here because it sounds like it has a similar meaning to "name" from `+name`.

This is not that important for me (compared to other suggestions I've made). Maybe it's just me trying to do unneeded stuff. Feel free to reject this proposal.
